lib/adagrams.rb

```diff
@@ -0,0 +1,139 @@
+require 'csv'
+
+def draw_letters
+  letters = {
+    "A" => 9,
+    "B" => 2,
+    "C" => 2,
+    "D" => 4, 
+    "E" => 12, 
+    "F" => 2,
+    "G" => 3, 
+    "H" => 2,
+    "I" => 9, 
+    "J" => 1,
+    "K" => 1,
+    "L" => 4,
+    "M" => 2,
+    "N" => 6,
+    "O" => 8,
+    "P" => 2,
+    "Q" => 1,
+    "R" => 6,
+    "S" => 4,
+    "T" => 6,
+    "U" => 4,
+    "V" => 2,
+    "W" => 2,
+    "X" => 1,
+    "Y" => 2,
+    "Z" => 1 
+  }
+
+  keys_only = letters.keys
+  copy_of_letters = letters
+  hand = []
+
+  while hand.length < 10
+    tile =  keys_only.sample
+    tile_count = copy_of_letters["#{tile}"]
+    if tile_count > 0
+      copy_of_letters["#{tile}"] -= 1
+      hand.push(tile)
+    end
+  end
+
+  return hand
+end 
```

Comment:
Nice work on this method! This is a good strategy. Actually, although you're right that sometimes you need to make a copy of variables, in this case, you don't need `copy_of_letters`. You can replace `copy_of_letters` here with `letters` and the tests still pass! Let me know if you have questions on that!